tmail-backend/webadmin/webadmin-rate-limit/src/main/scala/com/linagora/tmail/webadmin/RateLimitPlanManagementRoutes.scala

```diff
@@ -0,0 +1,178 @@
+package com.linagora.tmail.webadmin
+
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+import java.util
+import java.util.Optional
+
+import com.linagora.tmail.rate.limiter.api.LimitTypes.LimitTypes
+import com.linagora.tmail.rate.limiter.api.{Count, DeliveryLimitations, LimitTypes, OperationLimitations, OperationLimitationsType, RateLimitation, RateLimitationPlanRepository, RateLimitingPlan, RateLimitingPlanCreateRequest, RateLimitingPlanId, RateLimitingPlanName, RateLimitingPlanNotFoundException, RateLimitingPlanResetRequest, RelayLimitations, Size, TransitLimitations}
+import com.linagora.tmail.webadmin.model.{GetAllRateLimitPlanResponseDTO, OperationLimitationsDTO, RateLimitationDTO, RateLimitingPlanCreateRequestDTO, RateLimitingPlanDTO, RateLimitingPlanResetRequestDTO}
+import javax.inject.Inject
+import org.apache.james.rate.limiter.api.AllowedQuantity
+import org.apache.james.util.DurationParser
+import org.apache.james.webadmin.Constants.SEPARATOR
+import org.apache.james.webadmin.Routes
+import org.apache.james.webadmin.utils.{ErrorResponder, JsonExtractor, JsonTransformer, Responses}
+import org.eclipse.jetty.http.HttpStatus.NOT_FOUND_404
+import reactor.core.scala.publisher.{SFlux, SMono}
+import spark.{Request, Response, Route, Service}
+
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+import scala.jdk.StreamConverters._
+
+class RateLimitPlanManagementRoutes @Inject()(planRepository: RateLimitationPlanRepository,
+                                              jsonTransformer: JsonTransformer) extends Routes {
+  private val PLAN_ID_PARAM = ":planId"
+  private val PLAN_NAME_PARAM = ":planName"
+  private val BASE_PATH = SEPARATOR + "rate-limit-plans"
+  private val CREATE_A_PLAN_PATH = BASE_PATH + SEPARATOR + PLAN_NAME_PARAM
+  private val UPDATE_A_PLAN_PATH = BASE_PATH + SEPARATOR + PLAN_ID_PARAM
+  private val GET_A_PLAN_PATH = BASE_PATH + SEPARATOR + PLAN_ID_PARAM
+  private val GET_ALL_PLAN_PATH = BASE_PATH
```

Comment:
Please with scala you can use string interpolation to make this readable

(`s"ac $dds"`)